kernel/src/main/java/org/kframework/compile/checks/CheckSmtLemmas.java

```diff
@@ -0,0 +1,49 @@
+package org.kframework.compile.checks;
+
+import org.kframework.definition.Module;
+import org.kframework.definition.Rule;
+import org.kframework.definition.Sentence;
+import org.kframework.kore.KApply;
+import org.kframework.kore.KRewrite;
+import org.kframework.kore.VisitK;
+import org.kframework.utils.errorsystem.KEMException;
+
+import java.util.Set;
+
+public class CheckSmtLemmas {
+    private final Set<KEMException> errors;
+    private final Module m;
+
+    public CheckSmtLemmas(Set<KEMException> errors, Module m) {
+        this.errors = errors;
+        this.m = m;
+    }
+
+    public void check(Sentence sentence) {
+        if ((sentence instanceof Rule) && sentence.att().contains("smt-lemma")) {
+            check((Rule) sentence);
+        }
+    }
+
+    private void check(Rule rule){
+        if (!(rule.body() instanceof KRewrite)) {
+            return;
+        }
+        new VisitK() {
+            @Override
+            public void apply(KRewrite k) {
+                super.apply(k.left());
+                super.apply(k.right());
+            }
+
+            @Override
+            public void apply(KApply k) {
+                if (!k.att().toString().matches(".*smtlib.*|.*smt-hook.*")) {
```

Comment:
Yeah you can use `k.att().contains(Att.SMTLIB())`

Reply:
I ended up using
```
if (m.productionsFor().isDefinedAt(k.klabel()) && !m.productionsFor().get(k.klabel()).get()
         .exists(p -> p.att().contains(Att.SMT_HOOK()) || p.att().contains(Att.SMTLIB()))) {
```
instead. The `k.att().contains(Att.SMTLIB())` doesn't work because the SMTLIB is technically in k.att().att()